pkg/network/tracer/offsetguess.go

```diff
@@ -736,22 +736,33 @@ func newEventGenerator(ipv6 bool) (*eventGenerator, error) {
 		return nil, err
 	}
 
-	var udp6Conn *net.UDPConn
-	if ipv6 {
-		linkLocal, err := getIPv6LinkLocalAddress()
-		if err != nil {
-			return nil, err
-		}
-
-		udp6Conn, err = net.ListenUDP("udp6", linkLocal)
-		if err != nil {
-			return nil, err
-		}
+	udp6Conn, err := getUDP6Conn(ipv6)
+	if err != nil {
+		return nil, err
 	}
 
 	return &eventGenerator{listener: l, conn: c, udpConn: udpConn, udp6Conn: udp6Conn, udpDone: udpDone}, nil
 }
 
+func getUDP6Conn(ipv6 bool) (*net.UDPConn, error) {
+	if !ipv6 {
+		return nil, nil
+	}
+
+	if !kernel.IsIPv6Enabled() {
+		// TODO: Find a offset guessing method that doesn't need an available IPv6 interface
+		log.Debug("unable to find ipv6 device for udp6 flow offset guessing. unconnected udp6 flows won't be traced.")
```

Comment:
Just unconnected udp6 flows, or any udp6 flows?

Reply:
As far as I understand, in UDP sockets that are used after a `connect` syscall, the normal IPv6 fields (the ones we also rely on for TCPv6 traffic will be populated). So yes, only unconnected udp6.
@brycekahle please confirm if my understanding is correct 😅

Reply:
Can we include the error in the debug message?